app/validators/file_type_allowed_validator.rb

```diff
@@ -2,15 +2,19 @@ class FileTypeAllowedValidator < ActiveModel::EachValidator
   # The list of allowed types is security sensitive because it may allow files with script
   # capabilities to be uploaded leading to XSS attacks. Please be sure to know what you are
   # doing when modifying this list.
-  VALID_FILE_EXTENSIONS = [".jpg", ".jpeg", ".pdf", ".png", ".heic", ".bmp", ".txt", ".tiff", ".gif"]
-  VALID_MIME_TYPES = ["image/jpeg", "image/png", "application/pdf", "image/heic", "image/bmp", "text/plain", "image/tiff", "image/gif"]
+
+  # TODO: remove these
+  # VALID_FILE_EXTENSIONS = [".jpg", ".jpeg", ".pdf", ".png", ".heic", ".bmp", ".txt", ".tiff", ".gif"]
+  # VALID_MIME_TYPES = ["image/jpeg", "image/png", "application/pdf", "image/heic", "image/bmp", "text/plain", "image/tiff", "image/gif"]
 
   def validate_each(record, attr_name, value)
     return if !value
 
     extension = File.extname(value.path)
-    unless VALID_FILE_EXTENSIONS.include?(extension.downcase)
-      record.errors.add(attr_name, I18n.t("validators.file_type"))
+    valid_types_from_record = record.classify.constantize::VALID_FILE_TYPES
```

Comment:
`record.classify.constantize::VALID_FILE_TYPES` is some good Ruby evil genius programming; I do wonder if there's a simpler way to do it. Can you do `record.class::VALID_FILE_TYPES`?

Reply:
The old code used to validate file _extensions_ but this code seems to validate _mime types_. I'm not sure the new approach works.